### PR TITLE
Support 2FA with OAuth2 logging in

### LINKED ISSUE #2939: OAuth login skips TFA
### Describe the issue

When logging in with an OAuth account, if the NamelessMC user has tfa enabled, it gets skipped.

### NamelessMC Version

Development version

### To Reproduce

_No response_

### Expected Behaviour

We should complete the OAuth flow in the `modules/Core/pages/oauth.php`, and then instead of logging in and redirecting to the homepage we should redirect to the tfa page.

### Screenshots

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by supercrafter100**
Closes #2939

### CODE REVIEW

File: modules/Core/pages/oauth.php [hunk 1]
```diff
@@ -10,6 +13,31 @@
     throw new RuntimeException('No OAuth method set');
 }
 
+// If they are filling in 2FA. We've already retrieved their user. We can skip the other steps in this case
+if (isset($_SESSION['user_id']) && isset($_POST['tfa_code'])) {
+    $user = new User($_SESSION['userid']);
```
Comment (partydragen) [inline modules/Core/pages/oauth.php:18]: if isset user_id but New User using userid????

**Review by partydragen — CHANGES_REQUESTED** [review]

File: modules/Core/pages/oauth.php (hunk 1 above)
Comment (supercrafter100) [reply modules/Core/pages/oauth.php:18]: whooops

**Comment by supercrafter100** [comment]
Code may need some cleanup but it works

File: modules/Core/pages/oauth.php [hunk 2]
```diff
@@ -1,5 +1,8 @@
 <?php
 
+const PAGE = 'oauth';
+require_once(ROOT_PATH . '/core/templates/frontend_init.php');
```
Comment (Aberdeener) [inline modules/Core/pages/oauth.php:4]: Why do we need frontend_init here?

Comment (supercrafter100) [reply modules/Core/pages/oauth.php:4]: So the template variable is properly assigned when the 2FA page is rendered

**Review by Aberdeener — APPROVED** [review]

**Review by samerton — APPROVED** [review]

**Review by partydragen — APPROVED** [review]

**Pull request merged**